igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/BuildController.groovy

```diff
@@ -226,20 +220,15 @@ class BuildController {
     @RequestMapping(value = '/builds/properties/{buildNumber}/{fileName}/{master:.+}/**')
     Map<String, Object> getProperties(
         @PathVariable String master,
-        @PathVariable Integer buildNumber, @PathVariable String fileName, HttpServletRequest request) {
+        @PathVariable Integer buildNumber, @PathVariable
+            String fileName, HttpServletRequest request) {
         def job = (String) request.getAttribute(
             HandlerMapping.PATH_WITHIN_HANDLER_MAPPING_ATTRIBUTE).split('/').drop(6).join('/')
         if (buildMasters.filteredMap(BuildServiceProvider.JENKINS).containsKey(master)) {
             Map<String, Object> map = [:]
             try {
                 def jenkinsService = buildMasters.map[master]
-                String path = getArtifactPathFromBuild(jenkinsService, job, buildNumber, fileName)
-
-                if (path == null) {
-                    log.error("Unable to get igorProperties: Could not find build artifact matching requested filename '{}' on '{}' build '{}",
-                            kv("fileName", fileName), kv("master", master), kv("buildNumber", buildNumber))
-                    return map
-                }
+                String path = getArtifactPathFromBuild(jenkinsService, master, job, buildNumber, fileName)
```

Comment:
Was previously just returning an empty map if the artifact path is not found.

Reply:
hi, when I delpoy spinnaker on k8s, the igor component show error:

Caused by: retrofit.converter.ConversionException: org.simpleframework.xml.core.ValueRequiredException: Unable to satisfy @org.simpleframework.xml.ElementList(entry=, data=false, inline=true, name=job, type=void, required=true, empty=true) on field 'list' private java.util.List com.netflix.spinnaker.igor.jenkins.client.model.ProjectsList.list for class com.netflix.spinnaker.igor.jenkins.client.model.ProjectsList

can you give me some clues, I have try to solve this problem by build docker image from source code, but failed, thanks!

Reply:
Can you raise this in a relevant forum? Either the #dev or #kubernetes channels on the Spinnaker Slack would seem appropriate. This issue and this repository have absolutely nothing to do with Spinnaker's support for Kubernetes or any problems you might be experiencing with that.